### ISSUE TITLE
[Firefox] jquery-fileupload + filestyle = change event not triggered

### ISSUE THREAD

**Issue by munchj** [markusslima/bootstrap-filestyle#61]
Hello,

I'm having an issue on firefox (it works fine on other browsers) when using filestyle along with jquery-fileupload. It seems that the 'on change' event is only triggered the first time when selecting a file.

Here is a minimalistic script to reproduce the issue :

``` html
<html>
<header>
<title>Test bug fileupload ff</title>
<script type="text/javascript" src="jquery-2.1.0.min.js"></script>
<script type="text/javascript" src="jquery-ui-1.10.4.min.js"></script>
<script type="text/javascript" src="bootstrap-filestyle.min.js"></script>
<script type="text/javascript" src="jQuery.FileUpload/jquery.fileupload.js"></script>
</header>
<body>
<input type="file" id="msg_attachement" />
<script type="text/javascript">
$(function() {
            $("#msg_attachement").filestyle();

            $("#msg_attachement").fileupload({
                add: function (e, data) {
                    alert('add');
                    return e;
                }
            });
});
</script>
</body>
</html>
```

On firefox, the alert is only triggered once even after several file selection.
Disabling filestyle fixes the issue (at the cost of the styling obviously)

Do you have any idea what the problem could be ?

Best regards,
Julien

**Comment by gdebrion** [comment]
Hello,

One year after, still the same conflict. The solution of @dtamajon works well except it opens two file picker windows. Only the first one is working…

I don't understand if it's a problem relative to FileUpload or to FileStyle :(

**Comment by markusslima** [comment]
I never plugin fileupload so I do not know. When searching countless met with the same or similar. after a while I waited for someone inform me what the plugin that was presenting problem. if you are willing to help and give me the link of the plugin see if I can solve the problem with all pleasure

**Comment by gdebrion** [comment]
The plugin is this one: https://github.com/blueimp/jQuery-File-Upload

Let me know if I can help you :)

**Comment by markusslima** [comment]
You do not have a some online example for me to observe more quickly?

**Comment by gdebrion** [comment]
Hey,

Here is a JsFiddle: http://jsfiddle.net/gdebrion/a4ffupL5/29/
You can check one time with the "filestyle" class on the input media, and one time without.

Steps:
1. Pick a file
2. Observe the progress bar moving
3. Pick another file
4. The upload starts again… or not, it depends if filestyle is active or not.

Tests done with Firefox 42.0 on Linux Mint Debian Edition.

**Comment by jasonrastovski** [comment]
Hi there!

I'm not sure if anyone is still having this issue, but I recently came across it with the project I'm working on, and I'm using both the jQuery File Upload plugin in combination with this bootstrap filestyle plugin.

The code block that starts at line 280 (in version 1.2.1) is only required if you are supporting Firefox version 22 & lower.  The Firefox bug that the code block was working around was fixed in version 23 and up.

This is the code block that can be removed: 
`            // Check if browser is Firefox
            if (window.navigator.userAgent.search(/firefox/i) > -1) {
                // Simulating choose file for firefox
                _self.$elementFilestyle.find('label').click(function () {
                    _self.$element.click();
                    return false;
                });
            }
`
In my application, we are only supporting "newer" versions of Firefox, so I've removed that code block entirely and everything functions as expected.  

Jason

**Comment by lenh** [comment]
Removing the above fixed it for me also - thanks!
